src/langcheck/metrics/eval_clients/_swallow.py

```diff
@@ -0,0 +1,230 @@
+# Built with Meta Llama 3
+
+from __future__ import annotations
+
+from typing import Iterable
+
+from transformers import AutoTokenizer
+from vllm import LLM, SamplingParams
+
+from ..prompts._utils import get_template
+from ._base import EvalClient
+
+
+class SwallowEvalClient(EvalClient):
+    """EvalClient defined for the Swallow models.
+    Swallow models only support English and Japanese.
+    The default model is set to "tokyotech-llm/Llama-3-Swallow-8B-Instruct-v0.1".
+    For other models, refer to the model hub at <https://huggingface.co/tokyotech-llm>.
+    """
+
+    def __init__(
+        self,
+        model_name: str = "tokyotech-llm/Llama-3-Swallow-8B-Instruct-v0.1",
+        torch_dtype: str = "bfloat16",
+        device: str = "cuda",
+    ):
+        """
+        Initialize the Swallow evaluation client.
+
+        Args:
+            model_name: The name of the model to use.
+            torch_dtype: The torch dtype to use. torch.bfloat16 is recommended.
+            device: The device to load the model on.
+        """
+        self._model = LLM(
+            model=model_name,
+            max_model_len=8192,
+            dtype=torch_dtype,
+            tensor_parallel_size=1,
+            device=device,
+        )
+        self._tokenizer = AutoTokenizer.from_pretrained(model_name)
+        self._sampling_params = SamplingParams(
+            temperature=0.6,
+            top_p=0.9,
+            max_tokens=1000,
+            stop="<|eot_id|>",
+            skip_special_tokens=True,
+        )
```

Comment:
The temperature and top_p are from https://huggingface.co/tokyotech-llm/Llama-3-Swallow-8B-Instruct-v0.1. The max_tokens in the demo is 512, but since step-by-step prompts lead to long generation, I increased it to about twice.

Reply:
Ah got it, sounds reasonable!